packages/http-service/src/stream.js

```diff
@@ -0,0 +1,27 @@
+import Rx from "rxjs";
+
+import ConnectionManager from "connection-manager";
+import { XHRConnection, ConnectionEvent } from "connections";
+
+export default function stream(url, options = {}) {
+  return Rx.Observable.create(function(observer) {
+    const connection = new XHRConnection(url, options);
+
+    connection.addListener(ConnectionEvent.DATA, function(event) {
+      observer.next(event.target.response);
```

Comment:
Let's always pass the event along and rather extend the event to also include the response.

Reply:
On one hand that totally makes sense, but then I was designing it to be a certain level of abstraction, so on `next` you just get the data.

Reply:
Following that logic we should adjust error and pass along only the error message, shouldn't we?

Reply:
That was my idea, but since we don't have a clear `errorMessage` field and `ERROR` event has so many faces I opted in for `event` in this case.

Reply:
I'd love to always pass the same type if possible.

Reply:
That means we would need to always map the happy path to unwrap the event.

Reply:
How about extending the connection event which always sends the data along?

Reply:
Shall it also send xhr.status? ...and the status code? :)

Reply:
We're going to pass an abstraction instead of the event in case of an error and will keep the _next_ as is, as discussed offline.

Reply:
+1 `observer.next({data: event.target.data})` – it would be in line with the error case, why would we need `type` here?!

IMHO in the future we should pass `DataConnectionEvent` and `ErrorConnectionEvent` to the callback/subscriber. As both are TBD yet, we should at least pass an object to be able to exchange them later on.